app/services/recovery_code_generator.rb

```diff
@@ -1,6 +1,8 @@
 class RecoveryCodeGenerator
   attr_reader :user_access_key
 
+  INVALID_CODE = 'this is an invalid recovery code but a valid string'.freeze
```

Comment:
not a typo.

I was cheating a bit. If you hand `Base32::Crockford.decode` an un-decodable string like `this string has U in it` then it will return nil, which the subsequent `encode` will throw an exception on. But I wanted `normalize_code` to always return a string, even if it was a bad string. So I created a constant that could never be a recovery code but was a valid string. The point is to trigger the decryption exception in `verify` so that the return value of `verify` is `false`.